### DISCUSSION TITLE
Vote: what's the best way to run the tutorials?

### DISCUSSION THREAD

**Discussion by javabster** [Qiskit/.github#6] [Documentation]
There are bunch of ways to work through the Qiskit Tutorials, let us know how you most enjoy working through them:

- In IBM Quantum Labs (👍)
- Reading through them in the [API reference website](https://qiskit.org/documentation/tutorials.html) (🚀 )
- Pulling from the tutorials repo and running locally (❤️ )
- Other - please comment!

**Comment by 1ucian0** [comment]
The textbook has a `Run` bottom that I love (like [here](https://learn.qiskit.org/course/ch-states/the-atoms-of-computation#first-circuit))
![Screenshot 2021-10-12 at 11 47 00](https://user-images.githubusercontent.com/766693/136933323-3f132a54-d024-4351-a93d-0d654417f4a5.png)

I'm not sure what's the magic behind them (maybe [Thebe](https://thebe.readthedocs.io/en/latest/)?)

**Reply by jwoehr** [reply]
> The textbook has a `Run` bottom that I love 

"Run it here" is the best for a tutorial.

But it should always be easy to download a notebook to one's own workstation, where playing around is the best way to deepen one's knowledge once one has grasped the basics from the tutorial.

**Reply by y4izus** [reply]
From my point of view, we should have in mind no just with which one do we enjoy the most, but where we can run the code. For example, right now the the difference between the code we can run on the textbook and the one on labs is that the second one use the real quantum computer (a login is needed). I mean, if the tutorial is talking about how to run code on a real device, we need (at this moment) labs.
I really like the idea of having the notebook but I also see the good part of having the code directly to run it on the web.

Said that, I like this combination:
- Use the textbook format as much as we could. No just for unify the learning resources for a better user experience, but also to take advantage of the features the textbook already has (metrics analytics, translations, ...)
- Add the option to download the notebook ( note that the textbook is built through notebooks that are converted into html, so we don't need extra work. See the example with the Grover chapter: [notebook](https://github.com/qiskit-community/platypus/blob/main/notebooks/ch-algorithms/grover.ipynb) and [html](https://learn.qiskit.org/course/ch-algorithms/grovers-algorithm) )
- Use labs for the tutorials that require a real device. Have also in mind that the digital team is working on a login feature, so maybe in the future we could also run these tutorials with the textbook format

**Reply by 1ucian0** [reply]
The @y4izus makes total sense to me.
